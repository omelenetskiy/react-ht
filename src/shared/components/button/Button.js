import * as React from 'react';
import StyledButton from '../../styled/button';

type Props = {
  size?: string,
  children?: React.Node,
  onClick: SyntheticEvent<>,
};

const Button = ({ children, onClick, size }: Props) => (
  <StyledButton size={size} onClick={onClick}>
    {children}
  </StyledButton>
);

Button.defaultProps = {
  size: 'small',
  children: null,
};

export default Button;
