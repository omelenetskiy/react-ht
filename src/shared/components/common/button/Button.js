import React from "react";

const Button = () => <button className="btn">button</button>;

export default Button;
