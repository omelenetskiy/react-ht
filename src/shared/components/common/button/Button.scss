.button {
  border: 0;
  text-transform: uppercase;
  padding: 7px 15px;
  color: #fff;
  background-color: $danger-color;
  font-weight: bold;
  border-radius: 3px;
  cursor: pointer;
  outline: none;

  &:hover {
    background-color: $danger-color-dark;
  }
}

.buttonOutline {
  background-color: transparent;
  padding: 50px;
}
