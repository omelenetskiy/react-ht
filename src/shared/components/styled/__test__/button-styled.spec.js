import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';

import Button from '../button';

const theme = {
  dangerColor: 'dangerColor',
  dangerColorDark: 'dangerColorDark'
};

const props = {
  size: 'small'
};

describe('Button styled', () => {
  it('should render correctly', () => {
    const wrapper = <Button theme={theme} {...props} />;

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
