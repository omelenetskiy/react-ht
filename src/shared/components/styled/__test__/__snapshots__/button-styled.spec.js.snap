// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styled should render correctly 1`] = `
.c0 {
  border: 0;
  padding: 7px 15px;
  color: #fff;
  background-color: dangerColor;
  font-weight: bold;
  border-radius: 3px;
  cursor: pointer;
  outline: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
}

.c0:hover {
  background-color: dangerColorDark;
}

<button
  className="c0"
  size="small"
/>
`;
